crowbar_framework/app/models/service_object.rb

```diff
@@ -708,7 +708,7 @@ def validate_proposal_constraints(proposal)
   def validate_one_for_role(proposal, role)
     elements = proposal["deployment"][@bc_name]["elements"]
 
-    if not elements.has_key?(role) or elements[role].length != 1
+    if not elements.key?(role) or elements[role].length != 1
```

Comment:
Style/AndOr: Use `||` instead of `or`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)<br>Style/Not: Use `!` instead of `not`. (https://github.com/bbatsov/ruby-style-guide#bang-not-not)